functions-python/helpers/database.py

```diff
@@ -18,15 +18,30 @@
 import threading
 from typing import Final
 
-from sqlalchemy import create_engine, text
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine, text, event
+from sqlalchemy.orm import sessionmaker, mapper
 import logging
 
 DB_REUSE_SESSION: Final[str] = "DB_REUSE_SESSION"
 lock = threading.Lock()
 global_session = None
 
 
+def set_cascade(mapper, class_):
+    if class_.__name__ == "Gtfsfeed":
+        for rel in class_.__mapper__.relationships:
+            if rel.key in [
+                "redirectingids",
+                "redirectingids_",
+                "externalids",
+                "externalids_",
+            ]:
+                rel.cascade = "all, delete-orphan"
+
+
+event.listen(mapper, "mapper_configured", set_cascade)
```

Comment:
Won't this be executed when the file is imported?Is this what you want?

Reply:
It will be executed when the SQLAlchemy triggers [the mapper configurated events](https://docs.sqlalchemy.org/en/13/orm/events.html). This is not exactly when the file is imported, it's part of the SqlAchemy lifecycle. It might happen once per thread while Sqlalchemy is instantiating the classes.